db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/StatementSchemaPermission.java

```diff
@@ -337,4 +360,35 @@ public String getSchemaName() {
 		return schemaName;
 	}
 
+	/**
+	 * Routine to check if another instance of access descriptor matches this.
+	 * Used to ensure only one access descriptor for a table of given privilege is created.
+	 * Otherwise, every column reference from a table may create a descriptor for that table.
+	 *
+	 * @param obj	Another instance of StatementPermission
+	 *
+	 * @return	true if match
+	 */
+	public boolean equals( Object obj)
+	{
+		if( obj == null)
+			return false;
+		if( getClass().equals( obj.getClass()))
```

Comment:
Why not `instanceOf`? Why are you comparing string values?

Reply:
I need to return true only if the two objects are exactly of the same class, no true for sub-super class relationship. It will be used to search/match entries in dictionary cache.